internal/rbd/controllerserver.go

```diff
@@ -188,31 +188,37 @@ func getGRPCErrorForCreateVolume(err error) error {
 	return status.Error(codes.Internal, err.Error())
 }
 
-// validateRequestedVolumeSize validates the request volume size with the
-// source snapshot or volume size, if there is a size mismatches it returns an error.
-func validateRequestedVolumeSize(rbdVol, parentVol *rbdVolume, rbdSnap *rbdSnapshot, cr *util.Credentials) error {
+// source snapshot or volume size, if there is a size missmatch and if the new volume is not block mode
+// it returns an error.
+func validateRequestedVolumeSize(rbdVol *rbdVolume, parentVol *rbdVolume, rbdSnap *rbdSnapshot, cr *util.Credentials, isBlock bool) (bool, error) {
```

Comment:
why we need a bool flag in return here? if we return bool from here we are doing two validation, can we skip this one as do validation in the caller itself?

Reply:
I do not get it either. The return value needs to be explained in the comment of the function. Please follow go-doc standards too.